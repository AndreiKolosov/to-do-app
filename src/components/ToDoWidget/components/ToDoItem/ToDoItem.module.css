.item {
  display: flex;
  justify-content: space-between;
  border: 1px solid rgb(97, 155, 155);
  border-radius: 10px;
  padding: 5px 10px;
}

.item__inputLabel {
  display: grid;
	align-items: center;
	max-width: max-content;
	cursor: pointer;
	grid-template-columns: 20px auto;
	gap: 0 20px;
  margin: 0;
	text-align: left;
	color: #000;
	font-family: Verdana, Geneva, Tahoma, sans-serif;
	font-size: 1rem;
	line-height: 1;
}

.item__inputLabel_checked {
  position: relative;
  opacity: 0.5;
}

.item__inputLabel_checked::after {
  position: absolute;
  content: '';
  left: 0;
  top: 50%;
  width: 100%;
  height: 1px;
  background-color: #000;
}

.item__input {
  width: 20px;
	height: 20px;
	padding: 0;
	cursor: pointer;
	border-color: transparent;
	background-image: url("../../../../images/icons/checkbox.svg");
	background-repeat: no-repeat;
	background-position: center;
	background-size: contain;
	appearance: none;
}

.item__input:checked {
	background-image: url("../../../../images/icons/checkbox-checked.svg");
}

.item__button {
  border: 1px solid rgb(121, 119, 119);
  background-color: #dd9898;
  cursor: pointer;
  border-radius: 8px;
  padding: 0 10px;
  line-height: 1;
}