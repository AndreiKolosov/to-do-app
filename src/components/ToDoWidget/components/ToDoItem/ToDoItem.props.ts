import { DetailedHTMLProps, HTMLAttributes } from 'react';
import { TToDoItem } from '../../../../utils/types';
export interface IToDoItemProps extends DetailedHTMLProps<HTMLAttributes<HTMLLIElement>, HTMLLIElement> {
  itemData: TToDoItem;
  changeHandler: (id: string) => void;
  removeHandler: (id: string) => void;
}