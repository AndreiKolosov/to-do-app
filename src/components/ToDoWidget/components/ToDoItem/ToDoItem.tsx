import cx from 'classnames';
import { FC } from 'react';
import styles from './ToDoItem.module.css';
import { IToDoItemProps } from './ToDoItem.props';

const ToDoItem: FC<IToDoItemProps> = ({ itemData, removeHandler, changeHandler, className, ...props }) => (
  <li className={cx(styles.item, className)} {...props}>
    <label
      htmlFor={itemData.id}
      className={cx(styles.item__inputLabel, {
        [styles.item__inputLabel_checked]: itemData.completed,
      })}
    >
      <input
        className={styles.item__input}
        checked={itemData.completed}
        type='checkbox'
        onChange={() => changeHandler(itemData.id)}
        name={itemData.label}
        id={itemData.id}
      />
      {itemData.label}
    </label>
    <button type='button' className={styles.item__button} onClick={() => removeHandler(itemData.id)}>
      X
    </button>
  </li>
);

export default ToDoItem;
