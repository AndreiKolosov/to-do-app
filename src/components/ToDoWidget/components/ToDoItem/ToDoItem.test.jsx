import renderer from 'react-test-renderer';
import {render, screen} from '@testing-library/react';

import ToDoItem from './ToDoItem';

const itemData = { label: 'Дело', completed: true, id: 'asd' };

describe('ToDo Item Component', () => {
  it('Item render', () => {
    render(<ToDoItem itemData={itemData} removeHandler={() => {}} changeHandler={() => {}} />);
    expect(screen.getByText('Дело')).toBeInTheDocument();
  });

  it('To-do item рендерится без ошибок',() => {
      const tree = renderer
        .create(<ToDoItem itemData={itemData} removeHandler={() => {}} changeHandler={() => {}} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
})

