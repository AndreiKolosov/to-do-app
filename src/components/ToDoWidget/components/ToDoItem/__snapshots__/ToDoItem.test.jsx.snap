// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDo Item Component To-do item рендерится без ошибок 1`] = `
<li
  className="item"
>
  <label
    className="item__inputLabel item__inputLabel_checked"
    htmlFor="asd"
  >
    <input
      checked={true}
      className="item__input"
      id="asd"
      name="Дело"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="item__text"
    >
      Дело
    </span>
  </label>
  <button
    className="item__button"
    onClick={[Function]}
    type="button"
  >
    X
  </button>
</li>
`;
